capa/features/common.py

```diff
@@ -92,23 +97,32 @@ def __nonzero__(self):
 
 
 class Feature(abc.ABC):
-    def __init__(self, value: Union[str, int, float, bytes], description=None):
+    def __init__(
+        self, value: Union[str, int, float, bytes], access: Optional[str] = None, description: Optional[str] = None
```

Comment:
why does the `Feature` superclass get the `access` property? on first read, it doesn't seem like `access` applies to some of the subclasses, like `MatchedRule`, so it feels a little weird here.

I dont really want to spent a lot of time/effort debating class heirarchies and overengineering here, so feel free to ignore if there's a good reason for this.

Reply:
what are the features that should support `access`?

from memory:
  - number
  - string
  - property

Reply:
`Feature` was extended with `access` to make it simpler to add this to other subclasses in the future. This PR adds `access` support for `property` but as you pointed out other candidates for future PRs include `number`, `offset`, and possibly `string`.

Reply:
I've added a subclass of `Feature` called `_AccessFeature` from which features like `property`, `number`, etc. can derive `access`.